### PR TITLE
Build and push multi-arch binaries and images for amd64 and arm64

### PR DESCRIPTION
**Pull request by ozangunalp**
Build workflow released snapshot images using: https://github.com/ozangunalp/kafka-native/actions/runs/8102824703

### CODE REVIEW

**Comment by ozangunalp** [comment]
@k-wall would you mind double-checking the workflows? I've run the build and copied it to the others.

**Comment by k-wall** [comment]
Sorry, I overlooked this, I'll take a look tomorrow

File: .github/workflows/build-kafka-images.yml [hunk 1]
```diff
@@ -11,57 +11,126 @@ on:
...
+        arch:
+          - amd64
+          - arm64
+    permissions:
+      contents: read
+      packages: write
+    name: Build Docker Images for ${{ matrix.arch }}
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
```
Comment (k-wall) [inline .github/workflows/build-kafka-images.yml:25]: Aside: dependabot has the ability to keep these things current. https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot

**Review by k-wall — APPROVED** [review]
looks reasonable to me.

**Pull request merged**